### DISCUSSION TITLE
On the behavior of the getLastModified() for the onlyNodeId argument.

### DISCUSSION THREAD

**Discussion by JanezStupar** [cachapa/sql_crdt#5] [General]
Hello Daniel,

I am contemplating the behavior of the `getLastModified` function, which seems unexpected at the moment whenever the CRDT receives an unknown nodeId for the `onlyNodeId` argument.

Assume the following test case:
```
    test('getLastModified for unknown node', () async {
      final result = await crdt.getLastModified(onlyNodeId: 'unknown_node');
      expect(result.nodeId, 'unknown_node');
    });
```

It fails because `sql_crdt` returns a zero HLC from its host node.

Arguably, the user can check whether the nodeId's match. But my intuition is that the zero ID should be for the requested nodeId.

Merry Christmas!

**Comment by cachapa** [comment]
The `modified` timestamps always use the host node id, so it makes sense that the zero HLC will also use that.

If a node id is unknown it makes sense that a zero HLC is returned since that's used for synchronization: new nodes will want to get the entire history rather than a subset.
